File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.js
function solution(stones, k) {
    let start = 1;
    let end = 200000000;
    
    while (start <= end) {
        let mid = Math.floor((start + end) / 2);
        let skip  = 0;
        
        for (let stone of stones) {
            if (stone < mid) skip++;
            else skip = 0;
            if (skip >= k) break;
        }
        
        if (skip >= k) end = mid - 1;
        else start = mid + 1;
    }
    
    return end;
}

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
import math

def solution(stones, k):
    answer = 0
    
    start = 1
    end = 200000000
    
    while start <= end:
        mid = math.floor((start + end) / 2)
        skip = 0
        
        for stone in stones:
            if stone < mid:
                skip += 1
            else:
                skip = 0
            if skip >= k:
                break
        
        if skip >= k:
            end = mid - 1
        else:
            start = mid + 1
               
    return end